cmd/dp-recipe-api/main.go

```diff
@@ -34,89 +53,111 @@ func main() {
 
 	log.Event(ctx, "config on startup", log.INFO, log.Data{"config": cfg})
 
+	versionInfo, healthErr := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+	if healthErr != nil {
+		log.Event(ctx, "failed to create service version information", log.ERROR, log.Error(healthErr))
+		os.Exit(1)
+	}
+
+	hc := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
 	//Feature Flag for Mongo Connection
 	enableMongoData := cfg.MongoConfig.EnableMongoData
 
-	if enableMongoData {
-		mongodb := &mongo.Mongo{
-			Collection: cfg.MongoConfig.Collection,
-			Database:   cfg.MongoConfig.Database,
-			URI:        cfg.MongoConfig.BindAddr,
-		}
+	mongodb := &mongo.Mongo{
+		Collection: cfg.MongoConfig.Collection,
+		Database:   cfg.MongoConfig.Database,
+		URI:        cfg.MongoConfig.BindAddr,
+	}
 
-		var err error
-		mongodb.Session, err = mongodb.Init()
-		if err != nil {
-			log.Event(ctx, "failed to initialise mongo", log.FATAL, log.Error(err))
-			os.Exit(1)
-		}
+	session, err := mongodb.Init()
+	if err != nil {
+		log.Event(ctx, "failed to initialise mongo", log.FATAL, log.Error(err))
+		os.Exit(1)
+	} else {
+		mongodb.Session = session
+		log.Event(ctx, "listening to mongo db session", log.INFO, log.Data{
+			"bind_address": cfg.BindAddr,
+		})
 	}
 
-	router := mux.NewRouter()
-	router.Methods("GET").Path("/health").HandlerFunc(healthcheck)
-	router.Methods("GET").Path("/recipes").HandlerFunc(recipeListHandler)
-	router.Methods("GET").Path("/recipes/{id}").HandlerFunc(recipeHandler)
+	mongoClient := mongoHealth.NewClient(session)
 
-	log.Event(ctx, "starting http server", log.INFO, log.Data{"bind_addr": cfg.BindAddr})
-	srv := server.New(cfg.BindAddr, router)
-	if err := srv.ListenAndServe(); err != nil {
-		log.Event(ctx, "error starting http server for API", log.FATAL, log.Error(err))
-		os.Exit(1)
+	mongoHealth := mongoHealth.CheckMongoClient{
```

Comment:
If the `enableMongoData` toggle is off, should we also prevent the healthcheck against MongoDB?

Reply:
That makes sense to me. If Mongo is not been used, then there is no point in adding a mongo health checker. Therefore, I have added the checker under the enableMongoData toggle. Done in PR https://github.com/ONSdigital/dp-recipe-api/pull/127